hydra-node/test/Hydra/API/HTTPServerSpec.hs

```diff
@@ -56,15 +54,9 @@ spec = do
 
     apiServerSpec
```

Comment:
Could: actually test the `/commit` endpoint response matching what we have written in the `api.yaml`

The `Validate /commit subscribe api schema` test above does not actually check the response matching a schema, but only that the `DraftCommitTxResponse` type encoding matches it.

We could easily convert this test to a wai-based one (like `GET /protocol-parameters` below) using a mocked chain handle which returns a test-data transaction on `draftCommitTx`.